src/cache/aeolus-cache.ts

```diff
@@ -0,0 +1,81 @@
+import { StoreCache } from '../interfaces/index'
+import { MissingStoreCahceError } from '../errors/index'
+
+export class AeolusCache {
+    private static instance: AeolusCache;
+    private storeCache?: StoreCache;
+
+    private constructor() {}
+
+    static getInstance() : AeolusCache {
+        if(!AeolusCache.instance) {
+            AeolusCache.instance = new AeolusCache();
+        }
+    
+        return AeolusCache.instance;
+    }
+
+    static setStoreCache(cache: StoreCache) {
+        let instance = this.getInstance();
+        instance.storeCache = cache;
+    }
+
+    count(key: string, period: number) : number {
+        if(this.storeCache){
+            var keyExpire : KeyExpire = this.keyAndExpiry(key, period);
+            return this.storeCache.increment(keyExpire.key, 1, keyExpire.expire);
+        } else {
+            throw new MissingStoreCahceError("Uninitialize store cache");
+        }
+    }
+
+    read(key: string) : number {
+        if(this.storeCache){
+            return this.storeCache.read(key);
+        } else {
+            throw new MissingStoreCahceError("Uninitialize store cache");
+        }
+    }
+
+    write(key: string, value: number, expiresIn: number) {
+        if(this.storeCache) {
+            this.storeCache.write(key, value, expiresIn);
+        } else {
+            throw new MissingStoreCahceError("Uninitialize store cache");
+        }
+    }
+
+    resetCount(key: string, period: number) {
+        if(this.storeCache) {
+            var keyExpire : KeyExpire = this.keyAndExpiry(key, period);
+            this.storeCache.delete(keyExpire.key);
+        } else {
+            throw new MissingStoreCahceError("Uninitialize store cache")
```

Comment:
This exception will be thrown in backed service, if no one catch it - it will crush the app which using this package.
I suggest you work with Promises. You may choose which promises syntax sugar you will use in your lib `async/await with throw`  or just pure Promises syntax and return `Promise.reject` here. But i think it will be much clear for consumer.